### PR TITLE
[games_menu] - Fix extract covers

### PR DESCRIPTION
**Pull request by ManiacVera**
Hi Swat,

Please give it a review, it no longer shows errors to me with have 138 games, I hope it doesn't to you either, it still needs to give it performance.

### CODE REVIEW

File: applications/games_menu/modules/menu.c [hunk 1]
```diff
@@ -1421,7 +1419,10 @@ bool SavePresetGame(PresetStruct *preset)
 		uint32 heap = HEAP_MODE_AUTO;
 		uint32 cdda_mode = CDDA_MODE_DISABLED;
 
-		char *preset_file_name = MakePresetFilename(GetDefaultDir(menu_data.current_dev), GetGamesPath(GetDeviceType(full_path_game)), sector_data.md5, "games_menu");
+		char *preset_file_name = (char *)malloc(100);
```
Comment (DC-SWAT) [inline applications/games_menu/modules/menu.c:1422]: Better use stack for such small temp buffers.

File: src/img/copy.c [hunk 2]
```diff
@@ -118,23 +118,19 @@ bool copy_image_memory_to_file(kos_img_t *image_source, const char *image_dest,
 		}
 		else
 		{
+			if (FileExists(image_dest))
+			{
+				fs_unlink(image_dest);
```
Comment (DC-SWAT) [inline src/img/copy.c:123]: You can call `fs_unlink` only without file checking. It will we faster.

File: src/img/utils.c [hunk 3]
```diff
@@ -18,7 +18,21 @@ bool pvr_is_alpha(const char *filename)
...
+		fs_close(pFile);
+		return false;
+	}
+
+	if (fsize > sector_size)
+	{
+		fsize = sector_size;
+	}
+
 	uint8 *data = (uint8 *)memalign(32, fsize);
```
Comment (DC-SWAT) [inline src/img/utils.c:36]: I'm sure you can allocate 512 bytes in the stack for it too. Just don't forget align attribute.
Also you can read just multiple by 32 bytes, so you can reduce size to header size aligned to 32 bytes.

Comment (ManiacVera) [reply src/img/utils.c:36]: Yes, I know, I only added it in case someone modified the GDI and installed an empty PVR.
If there is an empty PVR and the validation does not exist, when using read image, could it flag error or not?

Comment (DC-SWAT) [reply src/img/utils.c:36]: For checking empty file you can use FileSize > 0

Comment (ManiacVera) [reply src/img/utils.c:36]: Yes, but in this case I already read the image, it is better with fs_total, right?

Comment (DC-SWAT) [reply src/img/utils.c:36]: Yes!

Comment (ManiacVera) [reply src/img/utils.c:36]: So what do I do with this point?

Comment (DC-SWAT) [reply src/img/utils.c:36]: Aah, empty you mean transparent? Alpha channel check I see. Just use empty this PVR :) This is rare case.

Comment (ManiacVera) [reply src/img/utils.c:36]: No, empty image really empty, not transparent.
That's what I check in that step and then send whether it's transparent or not.
The transparent image happens to me in several games, that's why I set this condition.

Comment (DC-SWAT) [reply src/img/utils.c:36]: Ideally, this should be done simultaneously with the attempt to load the image. But if it is so difficult, leave as is, just use stack instead of memalign.

File: applications/games_menu/modules/menu.c [hunk 4]
```diff
@@ -1488,9 +1489,9 @@ bool SavePresetGame(PresetStruct *preset)
 		if (FileExists(isoldr_preset_file_name))
 		{
 			fs_unlink(isoldr_preset_file_name);
-		}
+		}		
 		CopyFile(preset_file_name, isoldr_preset_file_name, 0);
-
+
```
Comment (DC-SWAT) [inline applications/games_menu/modules/menu.c:1494]: Unnecessary changes

Comment (ManiacVera) [reply applications/games_menu/modules/menu.c:1494]: Surely it was a mistake of the finger, I mean unintentionally hehe.
I'm going to delete it better, because the tabulator does it automatically.
